### PR TITLE
Update rrule.pyi

### PR DESCRIPTION
**Pull request by glyph**
rrule.weekday(...) returns new rrule.weekday objects, not _common.weekday.

### CODE REVIEW

**Comment by glyph** [comment]
(Effectively a follow-up from #3154 :))

**Comment by glyph** [comment]
Thanks @srittau - will update.

File: third_party/2and3/dateutil/_common.pyi [hunk 1]
```diff
@@ -1,9 +1,11 @@
-from typing import Optional
+from typing import Optional, TypeVar, Type
+
+_T = TypeVar("_T")
 
 class weekday(object):
     def __init__(self, weekday: int, n: Optional[int] = ...) -> None: ...
 
-    def __call__(self, n: int) -> weekday: ...
+    def __call__(self: Type[_T], n: int) -> _T: ...
```
Comment (srittau) [inline third_party/2and3/dateutil/_common.pyi]: Sorry, that was an error on my part. The `Type` is nonsense. It's just `self: _T` as `self` has the same type as the `return value`. Thatswhy the tests fail.

Comment (glyph) [reply third_party/2and3/dateutil/_common.pyi]: No worries. Should have caught it locally :-P.  Let me try again…
